### ISSUE TITLE
fix(docs): restructure and renaming "How-to's" and "Examples and Tutorials" sections

### ISSUE THREAD

**Issue by bschrammIBM** [strongloop/loopback-next#2087]
## Description / Steps to reproduce / Feature proposal
https://loopback.io/doc/en/lb4/index.html
I want to rename the How-To's section to be more descriptive. How-To could refer to many topics in the LoopBack 4 docs. It's hard to define what a "How to" is, but this section appears to contain miscellaneous optional implementation tasks. So some ideas for section title and definitions are: 
- "Implementation Guides" - contains optional tasks to implement LoopBack 
-"Usage Guides" - contains optional tasks for using LoopBack in a runtime environment 
-"Optional Implementation Steps" - descriptive but a bit wordy!
- Other - any suggestions?

The next change is:
Move the `rpc-server` example from `Examples and Tutorials` into the new Implementation Guides section. Then rename `Examples and Tutorials` to be only `Tutorials`. This clarifies the Tutorial section as it will contain only actual tutorials. 

## Current Behavior
Examples and Tutorials contains only one example, which fits better as an optional implementation step. The Tutorials section would be stronger without it.
How-To's is not the best title for the section. 
## Acceptance Criteria
- [ ] Make "Examples and Tutorials" more distinguishable
  - [ ] tutorial: step-by-step instructions on how to do certain things
  - [ ] examples: a sample application for a particular feature(s)

_See [Reporting Issues](http://loopback.io/doc/en/contrib/Reporting-issues.html) for more tips on writing good issues_

**Comment by bschrammIBM** [comment]
@dhmlau @bajtos Can you please add your thoughts about this change to the issue?

**Comment by dhmlau** [comment]
@bschrammIBM , thanks for bringing it up. 

For `How-to's`, I'm fine with "How-to's" or other variations.   "Usage Guides" would be good too since it contains instructions for a particular usage / use case. 

For `Examples and Tutorials`, per our discussion, I think you'd like to distinguish:
- tutorial: step-by-step instructions on how to do certain things
- examples: a sample application for a particular feature(s)

+1 to make this distinction.  Could you please elaborate on what you'd like to see on the sidebar and the content pages?  We can/should add in the example-shopping app to the examples too.

**Comment by bajtos** [comment]
The proposals presented so far looks good to me. I don't have strong opinions on this, to be honest, and I am happy to follow your lead.

**Comment by jannyHou** [comment]
Acceptance Criteria:

- [ ] Rename "How-tos" to a more descriptive name "Usage Guides"
- [ ] Make "Examples and Tutorials" more distinguishable 
  - Proposal: 
    - tutorial: step-by-step instructions on how to do certain things
    - examples: a sample application for a particular feature(s)

**Comment by jannyHou** [comment]
Yeah I remember Raymond prefers to keep "how tos" as it is. Let's keep the discussion open.
@raymondfeng could you elaborate more about the reason? Thanks.

**Comment by dhmlau** [comment]
Closing as done.
